python/lsst/pex/exceptions/exceptions.cc

```diff
@@ -0,0 +1,155 @@
+    clsOverflowError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<RangeError> clsRangeError(mod, "RangeError", py::base<Exception>()); 
+    clsRangeError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<TimeoutError> clsTimeoutError(mod, "TimeoutError", py::base<Exception>()); 
+    clsTimeoutError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<TypeError> clsTypeError(mod, "TypeError", py::base<Exception>()); 
+    clsTypeError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<UnderflowError> clsUnderflowError(mod, "UnderflowError", py::base<Exception>()); 
+    clsUnderflowError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<DomainError> clsDomainError(mod, "DomainError", py::base<Exception>()); 
+    clsDomainError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<InvalidParameterError> clsInvalidParameterError(mod, "InvalidParameterError", py::base<Exception>()); 
+    clsInvalidParameterError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<LengthError> clsLengthError(mod, "LengthError", py::base<Exception>()); 
+    clsLengthError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<OutOfRangeError> clsOutOfRangeError(mod, "OutOfRangeError", py::base<Exception>()); 
+    clsOutOfRangeError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<LogicError> clsLogicError(mod, "LogicError", py::base<Exception>()); 
+    clsLogicError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<NotFoundError> clsNotFoundError(mod, "NotFoundError", py::base<Exception>()); 
+    clsNotFoundError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::class_<RuntimeError> clsRuntimeError(mod, "RuntimeError", py::base<Exception>()); 
+    clsRuntimeError.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>());
+
+    py::register_exception_translator([](std::exception_ptr p) {
+        try {
+            if (p) std::rethrow_exception(p);
+        } catch (const Exception &e) {
+            static py::object current_exception;
```

Comment:
I don't understand why this is static; is it just to allow it to be passed by non-const reference to `raiseLsstException`?  (Perhaps naively) it seems like passing it by moved value would be more appropriate.